### PR TITLE
REP-2650 Added the ability to bypass valkyrie authorization if a role exist in client's catalog.

### PR DESCRIPTION
**Pull request by wdschei**

### CODE REVIEW

File: repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/CullingBypassValkyrieAuthorizationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint + "/resources", method: "GET",
+                headers: [
+                        'content-type': 'application/json',
+                        'X-Auth-Token': fakeIdentityV2Service.client_token,
+                ],
+                defaultHandler: jsonResp
+        )
+
+        then: "check response"
+        mc.receivedResponse.code == "200"
```
Comment (dmnjohns) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/CullingBypassValkyrieAuthorizationTest.groovy]: Could we perhaps add an assertion that the fakeIdentityV2Service has not been interacted with?

File: repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/GetDeviceBypassValkyrieAuthorizationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        when: "a request is made against a device with Valkyrie set permissions"
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint + "/resource/" + deviceID, method: method,
+                headers: [
+                        'content-type': 'application/json',
+                        'X-Auth-Token': fakeIdentityV2Service.client_token,
+                ]
+        )
+
+        then:
+        mc.receivedResponse.code == "200"
```
Comment (dmnjohns) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/GetDeviceBypassValkyrieAuthorizationTest.groovy]: We can do the no interaction check here as well.

File: repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/TranslateRolesBypassValkyrieTest.groovy [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        when: "a request is made against a device with Valkyrie set permissions"
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint + "/account/permissions", method: "GET",
+                headers: [
+                        'content-type': 'application/json',
+                        'X-Auth-Token': fakeIdentityV2Service.client_token,
+                ]
+        )
+
+        then:
+        mc.receivedResponse.code == "200"
```
Comment (dmnjohns) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/TranslateRolesBypassValkyrieTest.groovy]: And here.

**Comment by dmnjohns** [comment]
A comment about the functional test, but otherwise, this looks good to me. :+1: 

Will merge after that comment is addressed.

File: repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/TranslateRolesBypassValkyrieTest.groovy (hunk 3 above)
Comment (jennyvo) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/TranslateRolesBypassValkyrieTest.groovy]: will add check

File: repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/CullingBypassValkyrieAuthorizationTest.groovy (hunk 1 above)
Comment (jennyvo) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/CullingBypassValkyrieAuthorizationTest.groovy]: add check for not interact with valkyrie service but actually it still interacts with idendity to verify token and get roles

Comment (dmnjohns) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/CullingBypassValkyrieAuthorizationTest.groovy]: Right, totally what I meant. Good call.

**Comment by dmnjohns** [comment]
IRC notifier is hanging again, but all tests pass. Merging.

**Pull request merged**